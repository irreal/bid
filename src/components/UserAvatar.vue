<template>
  <div class="user-avatar">
    <span>{{ userInitials }}</span>
  </div>
</template>

<script>
import { mapGetters } from "vuex";

export default {
  name: "BrandHeader",
  methods: {
    toggleMenu() {
      this.$store.commit("toggleMenu", false);
    }
  },
  computed: {
    ...mapGetters({ user: "getUser" }),
    username() {
      return this.user ? this.user.user_metadata.full_name : "";
    },
    userInitials() {
      var initials = this.username.match(/\b\w/g) || [];
      initials = (
        (initials.shift() || "") + (initials.pop() || "")
      ).toUpperCase();
      return initials ? initials : "N/A";
    }
  }
};
</script>

<style scoped lang="scss">
.user-avatar {
  background-color: orange;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  span {
    font-size: 32px;
    color: white;
  }
}
</style>
