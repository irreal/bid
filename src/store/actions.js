export default {
  updateUser: ({ commit }, payload) => {
    commit("setUser", payload.currentUser);
  }
};
