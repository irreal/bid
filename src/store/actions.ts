import { ActionTree } from 'vuex';
import State from './state';

const actions: ActionTree<State, State> = {
    //   updateColor({ commit }: ActionContext<State, State>, newColor: string): void {
    //     commit(COLOR, newColor);
    //   },
    };

export default actions;
