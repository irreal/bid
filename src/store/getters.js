export default {
  getUserStatus: state => !!state.user,
  getUser: state => state.user
};
