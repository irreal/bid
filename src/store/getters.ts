import { GetterTree } from 'vuex';
import State from './state';
import Plan from '@/models/plan';

const getters: GetterTree<State, State> = {
    plans(state: State): Plan[] {
      return state.plans;
    },
    // colour(state: State): string {
    // 	return state.color;
    // },
  };

export default getters;
