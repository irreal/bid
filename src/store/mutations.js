export default {
  setUser: (state, currentUser) => {
    state.user = currentUser;
  }
};
