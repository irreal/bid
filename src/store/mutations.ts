import { MutationTree } from 'vuex';
import State from './state';

const mutations: MutationTree<State> = {
    removePlans(state: State) {
      state.plans = [];
    },
    // color(state: State, newColor: string): void {
    // 	state.color = newColor;
    // }
  };

export default mutations;
