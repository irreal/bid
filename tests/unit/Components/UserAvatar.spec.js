import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import UserAvatar from "@/components/UserAvatar";
import Vuetify from "vuetify";
import Vue from "vue";
import initialState from "@/store/state";
import actions from "@/store/actions";
import getters from "@/store/getters";
import mutations from "@/store/mutations";
import userFixture from "../fixtures/user";

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuetify);

describe("UserAvatar component", () => {
  let state;

  const build = () => {
    const ua = shallowMount(UserAvatar, {
      localVue,
      store: new Vuex.Store({ state, actions, getters, mutations })
    });

    return {
      ua
    };
  };

  beforeEach(() => {
    state = { ...initialState };
  });

  it("renders the component", () => {
    // arrange
    const { ua } = build();
    // assert
    expect(ua.html()).toMatchSnapshot();
    expect(ua.html()).toContain("N/A");
  });

  it("renders the component with a logged in user", () => {
    // arrange
    state.user = userFixture;
    const { ua } = build();
    // assert
    expect(ua.html()).toMatchSnapshot();
    let initials = userFixture.user_metadata.full_name.match(/\b\w/g) || [];
    initials = (
      (initials.shift() || "") + (initials.pop() || "")
    ).toUpperCase();
    expect(ua.html()).toContain(initials);
  });
});
