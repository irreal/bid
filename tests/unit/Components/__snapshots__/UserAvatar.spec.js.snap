// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar component renders the component 1`] = `<span>N/A</span>`;

exports[`UserAvatar component renders the component with a logged in user 1`] = `<span>MS</span>`;
